adafruit_mcp3xxx/analog_in.py

```diff
@@ -26,40 +26,44 @@
 differential ADC readings.
 
 * Author(s): Brent Rubell
+
+.. warning::
+    The ADC chips supported by this library do not use negative numbers. If the resulting
+    differential read is less than 0, then the returned integer value (and voltage value) is ``0``.
+    If for some reason the voltage on a channel is greater than the reference voltage or
+    less than 0, then the returned integer value is ``65472‬`` or ``0`` respectively.
+
 """
 
+from .mcp3xxx import MCP3xxx
+
 class AnalogIn():
     """AnalogIn Mock Implementation for ADC Reads.
 
-    :param ~mcp3004.MCP3004,~mcp3008.MCP3008 mcp: The mcp object.
+    :param MCP3002,MCP3004,MCP3008 mcp: The mcp object.
     :param int positive_pin: Required pin for single-ended.
     :param int negative_pin: Optional pin for differential reads.
-
     """
     def __init__(self, mcp, positive_pin, negative_pin=None):
+        if not isinstance(mcp, MCP3xxx):
+            raise ValueError("mcp object is not a sibling of MCP3xxx class.")
         self._mcp = mcp
         self._pin_setting = positive_pin
-        self._negative_pin = negative_pin
-        self.is_differential = False
-        if negative_pin is not None:
-            self.is_differential = True
-            self._channels = []
-            try:
-                self._pins = self._mcp.MCP3008_DIFF_PINS
-            except AttributeError:
-                self._pins = self._mcp.MCP3004_DIFF_PINS
-            self._pin_setting = self._pins.get((self._pin_setting, self._negative_pin),
-                                               "Difference pin not found.")
-
-    def __getitem__(self, key):
-        return self._channels[self._pins[key]]
+        self.is_differential = negative_pin is not None
+        if self.is_differential:
+            self._pin_setting = self._mcp.DIFF_PINS.get((positive_pin, negative_pin), None)
+            if self._pin_setting is None:
+                raise ValueError("Differential pin mapping not defined. Please read the docs.")
 
     @property
     def value(self):
-        """Returns the value of an ADC pin as an integer."""
+        """Returns the value of an ADC pin as an integer. Due to 10-bit accuracy of the chip, the
+        returned values range [0, 65472]."""
         return self._mcp.read(self._pin_setting, is_differential=self.is_differential) << 6
 
     @property
     def voltage(self):
-        """Returns the voltage from the ADC pin as a floating point value."""
+        """Returns the voltage from the ADC pin as a floating point value. Due to the 10-bit
+        accuracy of the chip, returned values range from 0 to (``reference_voltage`` *
+        65472 / 65535)"""
```

Comment:
Wouldn't returned values be:

> returned values range from 0 to (``adc_pin_value`` *
        ``reference_voltage`` / 65535)"""

Reply:
I'm really trying to state just the max and min values, so using the `adc_pin_value` would mean `1023` which gets translated into `65472`.

Reply:
wouldnt the `value` of the ADC pin be the pin reading from 0 to 65,472?

Reply:
Yes and that's what I put in the docs for the `value` attribute, but this line (`voltage` attribute) depends on the `reference_voltage` passed to `MCP3xxx` c'tor

Reply:
Ah, ok, I misread. Looks good then.